src/query/query_asset_scope_attribute.rs

```diff
@@ -114,34 +115,69 @@ pub fn may_query_scope_attribute_by_asset_uuid<S: Into<String>>(
 pub fn may_query_scope_attribute_by_scope_address<S: Into<String>>(
     deps: &DepsC,
     scope_address: S,
-) -> AssetResult<Option<AssetScopeAttribute>> {
+) -> AssetResult<Option<Vec<AssetScopeAttribute>>> {
     let querier = ProvenanceQuerier::new(&deps.querier);
     let scope_address_str: String = scope_address.into();
-    // First, query up the scope in order to find the asset definition's type
-    let scope = querier.get_scope(&scope_address_str)?;
-    // Second, query up the asset definition by the scope spec, which is a unique characteristic to the scope spec
-    let asset_definition =
-        load_asset_definition_v2_by_scope_spec(deps.storage, scope.specification_id)?;
-    // Third, construct the attribute name that the scope attribute lives on by mixing the asset definition's asset type with state values
-    let attribute_name = asset_definition.attribute_name(deps)?;
-    // Fourth, query up scope attributes attached to the scope address under the name attribute.
-    // In a proper scenario, there should only ever be one of these
-    let scope_attributes = querier.get_json_attributes::<_, _, AssetScopeAttribute>(
-        Addr::unchecked(&scope.scope_id),
-        &attribute_name,
-    )?;
-    // This is a very bad scenario - this means that the contract messed up and created multiple attributes under
-    // the attribute name.  This should only ever happen in error, and would require a horrible cleanup process
-    // that manually removed the bad attributes
-    if scope_attributes.len() > 1 {
-        return ContractError::generic(format!(
-            "more than one asset scope attribute exists at address [{}]. data repair needed",
-            &scope.scope_id
-        ))
-        .to_err();
+
+    // First, query up the scope to verify it exists
+    let _scope = querier.get_scope(&scope_address_str)?;
+
+    // Second, query up all possible asset definition names
+    let state = config_read_v2(deps.storage).load()?;
+    let asset_definitions: HashSet<String> = list_asset_definitions_v3(deps.storage)
+        .iter()
+        .map(|def| def.attribute_name_state(&state))
+        .collect();
+
+    // Third, query up asset scope attributes attached to the scope address under the name attribute.
+    // In a proper scenario, there should only ever be one of each type of these
+    let scope_attributes_v: Vec<(String, AssetScopeAttribute)> = querier
+        .get_attributes(Addr::unchecked(&scope_address_str), None::<String>)?
```

Comment:
Might be better off iterating over all asset attribute names and running individual queries for each.  That's obviously more blockchain action spam, but I could see a potential nightmare scenario where a scope has like an insane amount of attributes attached to it and you pick up like 100+ things here and then iterate over them.  What do you think?

Reply:
Hmm... good point... I wonder what the benefit of doing it that way for a potentially edge-case scenario vs. a maybe more common scenario of having two asset classes for a scope resulting in a couple queries 🤔

Reply:
You're absolutely right - it can have a bottleneck for either scenario based on the implementation, so your current code that caters to the more likely one is the way to go 👍